File: packages/cli/src/lib/paths.ts
import os from "node:os";
import path from "node:path";

/** ~/.config/agentblip (XDG_CONFIG_HOME respected). */
export function configDir(): string {
  const base = process.env.XDG_CONFIG_HOME || path.join(os.homedir(), ".config");
  return path.join(base, "agentblip");
}

/** ~/.local/state/agentblip (XDG_STATE_HOME respected). */
export function stateDir(): string {
  const base =
    process.env.XDG_STATE_HOME || path.join(os.homedir(), ".local", "state");
  return path.join(base, "agentblip");
}

export function configPath(): string {
  return path.join(configDir(), "config.json");
}

export function pidFilePath(): string {
  return path.join(stateDir(), "daemon.pid");
}

export function spawnLockPath(): string {
  return path.join(stateDir(), "daemon.spawn.lock");
}

export function logFilePath(): string {
  return path.join(stateDir(), "daemon.log");
}

File: apps/menubar/Sources/AgentblipMenuBar/Paths.swift
import Foundation

/// Resolves the same config/state locations as the CLI (packages/cli/src/lib/paths.ts),
/// honoring XDG_CONFIG_HOME / XDG_STATE_HOME so a customized daemon is still found.
enum Paths {
    private static var home: String {
        FileManager.default.homeDirectoryForCurrentUser.path
    }

    private static func envDir(_ key: String, default fallback: [String]) -> String {
        if let v = ProcessInfo.processInfo.environment[key], !v.isEmpty { return v }
        return ([home] + fallback).joined(separator: "/")
    }

    static var configDir: String {
        envDir("XDG_CONFIG_HOME", default: [".config"]) + "/agentblip"
    }

    static var stateDir: String {
        envDir("XDG_STATE_HOME", default: [".local", "state"]) + "/agentblip"
    }

    static var configPath: String { configDir + "/config.json" }
    static var daemonSecretPath: String { stateDir + "/daemon.secret" }
}
